docs/diagrams/tag-untag/TagActivity.puml

```diff
@@ -0,0 +1,28 @@
+@startuml
+start
+:User executes tag command;
+
+if () then ([Invalid, throws ParseException])
+else ([Valid])
+    :Initialise TagCommand;
+    if () then ([Some Tag(s) don't exist])
+        while ()
+            -> [Some Tag(s) unchecked];
+            if () then ([Tag doesn't exist])
+                :            Add specified Tag to Tag set            ;
+            else ([Tag exist])
+
+            endif
+        endwhile
+        -> [All tags checked];
```

Comment:
The label for this conditional should be nearer to the start of the diamond for readability, although this could be a restriction of puml.